### PR TITLE
Fix logic to open item in Finder/explorer

### PR DESCRIPTION
**Pull request by rafeca**
Closes https://github.com/desktop/desktop/issues/9698

## Description

This PR fixes the behaviour of opening a repository in Finder (in macOS) to open the actual repository directory instead of its parent folder.

The bug was introduced in https://github.com/desktop/desktop/commit/edde78b3e05ccb2fb0ee2238bd3656e02367285f (I cannot find its associated PR, so maybe it was committed directly to `master`).

`openDirectorySafe()` was created in https://github.com/desktop/desktop/pull/2968 as a refactor of https://github.com/desktop/desktop/pull/2861, and from both the description of the PRs and my understanding of its logic, that function should be always called (both for macOS and Windows, since the function already has logic to handle both platforms).

More specifically, based on [this comment](https://github.com/electron/electron/issues/10095#issuecomment-317285391) we should always use `openItem()` to open a directory, we just happen to have some specific code in `openDirectorySafe()` to open the folder and focus Finder.

## Release notes

Notes: [Fixed] Open the repository folder when clicking on "Show in Finder" in macOS

### CODE REVIEW

**Comment by say25** [comment]
With this in mind, I think this will solve the issue I was running into on #9499. Ill try to get that up-to-date in the next few days.

**Comment by niik** [comment]
@outofambit Since I wrote a lot of this way back when it would be wonderful if you could do a review of this before the end of the month. Let me know if you don't think you'll have the time and me and @rafeca can do a paired review

File: app/src/lib/app-shell.ts [hunk 1]
```diff
@@ -7,8 +7,31 @@ export interface IAppShell {
   readonly moveItemToTrash: (path: string) => boolean
   readonly beep: () => void
   readonly openExternal: (path: string) => Promise<boolean>
+  /**
+   * Reveals the specified file using the operating
+   * system default application.
+   * Do not use this method with non-validated paths.
+   *
+   * @param path - The path of the file to open
```
Comment (alayoon) [inline app/src/lib/app-shell.ts:15]: 15

**Comment by niik** [comment]
@rafeca I started reviewing this and found some things that weren't ideal and started down another path of being even more cautious. I figure we could pair review this tomorrow if you have time.

**Comment by niik** [comment]
@tierninho Would love for you to take another spin with this now.

**Pull request merged**